### PR TITLE
refactor(quick-match): Extract quick-match team-standing derivation logic out of GetQuickMatchUseCase into the domain

### LINKED ISSUE #133: [Refactor]: Extract quick-match team-standing derivation logic out of GetQuickMatchUseCase into the domain
## What part of the code is being refactored?

`src/modules/quick_match/application/use_cases/get_quick_match_use_case.py:115-161` (`_compute_standing`) reconstructs "team A vs team B" rosters from `QuickMatchParticipant.team` for scoring purposes directly inside the use case, including a SINGLES-specific fallback (`team_b_ids` empty + exactly 2 participants → treat `participants[0]`/`participants[1]` as the two sides, lines 127-129).

## Motivation

Found during a follow-up Clean Architecture/DDD review of the features merged since the original backend audit (#106): friends system (#115), quick match module (#116, #117, #119, #121), and avatar (#125).

This is a business rule about how `QuickMatch` team structure maps to the two-sided scoring model — it duplicates knowledge already encoded once in `QuickMatch.add_participant()`'s team-assignment logic (`uses_teams`, "A" default), now with a second, ad-hoc implementation here. This is the same correctness class of bug that the M1/#110 fix (SINGLES WHS differential logic living in a use case instead of the domain) was meant to prevent, applied to a different rule.

Proposed fix: add a method on `QuickMatch` (or a small domain helper reusing `ScoringService`) — e.g. `team_rosters() -> tuple[set[ParticipantId], set[ParticipantId]] | None` — and call it from the use case instead of recomputing team membership inline.

## Risk / impact on existing tests

Low-medium — the resulting rosters should be identical, only where the derivation happens changes. Existing quick-match scoring/standing tests should catch any behavioral drift during the move; add unit tests for the new domain method directly, including the SINGLES fallback case.

## Priority

P2 — Normal.

## Effort estimate

**S (~half a day).** Extract `_compute_standing`'s team-derivation logic into a domain method mirroring the existing `QuickMatch`/`ScoringService` conventions, swap the use case to call it, and add direct unit tests for the new method (including the SINGLES fallback edge case) rather than only exercising it through `GetQuickMatchUseCase`.

### PR DESCRIPTION
**Pull request by agustinEDev**
## Summary
- `GetQuickMatchUseCase._compute_standing` reconstructed "team A vs team B" rosters from `QuickMatchParticipant.team` directly inside the use case, including a SINGLES-specific fallback — duplicating knowledge already encoded once in `QuickMatch.add_participant()`'s team-assignment logic.
- Added `QuickMatch.team_rosters() -> tuple[set[ParticipantId], set[ParticipantId]] | None`, mirroring the existing team-assignment convention (team field for FOURBALL/FOURSOMES, 2-participant fallback for SINGLES).
- `_compute_standing` now calls `quick_match.team_rosters()` instead of recomputing team membership inline.
- Added direct unit tests for `team_rosters()` on the domain entity (SINGLES fallback, FOURBALL from `team` field, <2 participants → `None`), independent of `GetQuickMatchUseCase`'s DTO/UoW plumbing.

Closes #133

## Test plan
- [x] `pytest tests/unit/modules/quick_match/` — 168 passed (165 + 3 new)
- [x] `ruff check` / `ruff format` — clean
- [x] `mypy` — no issues

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved quick-match standings by reliably identifying team rosters across singles and fourball matches.
  * Added support for two-player singles matches when determining opposing teams.
  * Prevented standings from appearing when valid team rosters cannot be determined, including incomplete team configurations.

* **Tests**
  * Added coverage for singles, fourball, missing-participant, and incomplete-roster scenarios to help ensure accurate standings.

### CODE REVIEW

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

File: src/modules/quick_match/domain/entities/quick_match.py [hunk 1]
```diff
@@ -349,6 +349,32 @@ def is_roster_complete(self) -> bool:
...
+        ejemplo, con menos de 2 participantes).
+        """
+        if len(self._participants) < 2:  # noqa: PLR2004
+            return None
+
+        team_a_ids = {p.participant_id for p in self._participants if (p.team or "A") == "A"}
+        team_b_ids = {p.participant_id for p in self._participants if p.team == "B"}
+        if not team_b_ids and len(self._participants) == 2:  # noqa: PLR2004 - SINGLES: no team field
+            team_a_ids = {self._participants[0].participant_id}
+            team_b_ids = {self._participants[1].participant_id}
```
[the remark this answers is not in the document]
Comment (agustinEDev) [reply src/modules/quick_match/domain/entities/quick_match.py]: Fixed — restricted the two-participant fallback to `self._match_format == MatchFormat.SINGLES` explicitly (matching your proposed diff), added the regression test for an incomplete FOURBALL with two participants on team A returning `None`, and added an explicit `match_format is None` guard for direct callers. 169/169 quick_match tests pass, full suite (2264) green.

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review

**Comment by agustinEDev** [comment]
@coderabbitai review
